pyteal/ast/opup.py

```diff
@@ -87,30 +94,27 @@ def __init__(self, mode: OpUpMode, target_app_id: Expr = None):
 
         self.mode = mode
 
+        require_type(inner_fee, TealType.uint64)
+        self.inner_fee = inner_fee
+
     def _construct_itxn(self) -> Expr:
         if self.mode == OpUpMode.Explicit:
-            return Seq(
-                InnerTxnBuilder.Begin(),
-                InnerTxnBuilder.SetFields(
-                    {
-                        TxnField.type_enum: TxnType.ApplicationCall,
-                        TxnField.application_id: self.target_app_id,
-                    }
-                ),
-                InnerTxnBuilder.Submit(),
+            return InnerTxnBuilder.Execute(
+                {
+                    TxnField.type_enum: TxnType.ApplicationCall,
+                    TxnField.application_id: self.target_app_id,
+                    TxnField.fee: self.inner_fee,
+                }
             )
         else:
-            return Seq(
-                InnerTxnBuilder.Begin(),
-                InnerTxnBuilder.SetFields(
-                    {
-                        TxnField.type_enum: TxnType.ApplicationCall,
-                        TxnField.on_completion: OnComplete.DeleteApplication,
-                        TxnField.approval_program: ON_CALL_APP,
-                        TxnField.clear_state_program: ON_CALL_APP,
-                    }
-                ),
-                InnerTxnBuilder.Submit(),
+            return InnerTxnBuilder.Execute(
+                {
+                    TxnField.type_enum: TxnType.ApplicationCall,
+                    TxnField.on_completion: OnComplete.DeleteApplication,
+                    TxnField.approval_program: ON_CALL_APP,
+                    TxnField.clear_state_program: ON_CALL_APP,
+                    TxnField.fee: self.inner_fee,
+                }
```

Comment:
@cusma Opening a thread to respond to https://github.com/algorand/pyteal/pull/553#issuecomment-1269794621 so that we avoid cluttering _Conversation_ tab.  Can you let me know if you agree with the assessment?  If so, we'll resolve the thread and consider the concern addressed.

* I think you raise a good question.  
* I agree the parameter name can be confusing in the conversation's context.  Though I advocate leaving as is due to the backwards compatibility concern.

Reply:
👍 for backward compatability

Reply:
Let's preserve backward compatibility then. I couldn't think about a better name than `inner_fee` for the Inner Transactions fee generated by `OpUp`. Open to better suggestions if any.

Reply:
@cusma Resolving - Thanks for the discussion here RE:  backwards compatibility.